topic-operator/src/main/java/io/strimzi/operator/topic/OperatorAssignedKafkaImpl.java

```diff
@@ -308,7 +308,7 @@ private String generateReassignment(Topic topic, String zookeeper) throws IOExce
         final ProcessHelper.ProcessResult processResult = ProcessHelper.executeSubprocess(executeArgs);
         ProcessHelper.delete(topicsToMove);
         String json = forEachLineStdout(processResult, new ReassignmentLineParser());
-        return json;
+        return json == null ? "{\"error\": \"Reassign Partitions is not supported\"}" : json;
```

Comment:
Why not just return a failed Future from `OperatorAssignedKafkaImpl.changeReplicationFactor(Topic topic)`?